src/node_contextify.cc

```diff
@@ -432,7 +432,10 @@ class ContextifyContext {
     // false for vmResult.x = 5 where vmResult = vm.runInContext();
     bool is_contextual_store = ctx->global_proxy() != args.This();
 
-    if (!is_declared && args.ShouldThrowOnError() && is_contextual_store)
+    bool is_function = value->IsFunction();
+
+    if (!is_declared && args.ShouldThrowOnError() && is_contextual_store
+    && !is_function)
```

Comment:
Style: `&&` should go on the previous line.

I'm curious, what does this do for `function f() {}` vs. `var f = function()` vs. `this.f = function() {}`?